.github/workflows/hardware-smoke.yml

```diff
@@ -263,3 +263,61 @@ jobs:
           python3 ./scripts/bmc_rtt.py -n --openocd /opt/tenstorrent/bin/openocd-rtt
           echo "SMC RTT logs:"
           python3 ./scripts/smc_rtt.py -n --openocd /opt/tenstorrent/bin/openocd-rtt
+
+  metal-test:
+    needs: smoke-e2e-test
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - board: p100
+            runs-on:
+              - p100-jtag
+          - board: p100a
+            runs-on:
+              - p100a-jtag
+          - board: p150a
+            runs-on:
+              - p150a-jtag
+    runs-on: ${{ matrix.config.runs-on }}
+    container:
+      image: ghcr.io/tenstorrent/tt-metal/upstream-tests-bh:v0.58.0-rc26-5-g668476a4d2
+      volumes:
+        - /dev/hugepages-1G:/dev/hugepages-1G
+        - /dev/hugepages:/dev/hugepages
+      options: '--device /dev/tenstorrent --device /dev/bus/usb --privileged'
+    env:
+      ARCH_NAME: blackhole
+    defaults:
+      run:
+        shell: bash
+        working-directory: /home/user/tt-metal/
+    steps:
+      - name: Install Unzip
+        run: |
+          sudo apt update
+          sudo apt install -y unzip protobuf-compiler
```

Comment:
Could these dependencies be added to the container image? That would probably simplify some things.